packages/@aws-cdk-testing/framework-integ/test/custom-resources/test/aws-custom-resource/integ.aws-custom-resource.ts

```diff
@@ -2,90 +2,110 @@
-  resourceType: 'Custom::SSMParameter',
-  onUpdate: {
-    service: 'SSM',
-    action: 'getParameter',
-    parameters: {
-      Name: ssmParameter.parameterName,
-      WithDecryption: true,
-    },
-    physicalResourceId: PhysicalResourceId.fromResponse('Parameter.ARN'),
-  },
-  policy: AwsCustomResourcePolicy.fromSdkCalls({ resources: AwsCustomResourcePolicy.ANY_RESOURCE }),
-});
+    const customRole = new iam.Role(this, 'CustomRole', {
+      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+    });
+    customRole.addToPolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        resources: ['*'],
+        actions: [
+          'ssm:*',
+        ],
+      }),
+    );
+    const getParameterNoPolicy = new AwsCustomResource(this, 'GetParameterNoPolicy', {
+      resourceType: 'Custom::SSMParameter',
+      onUpdate: {
+        service: 'SSM',
+        action: 'getParameter',
+        parameters: {
+          Name: ssmParameter.parameterName,
+          WithDecryption: true,
+        },
+        physicalResourceId: PhysicalResourceId.fromResponse('Parameter.ARN'),
+      },
+      role: customRole,
+    });
 
-const customRole = new iam.Role(stack, 'CustomRole', {
-  assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
-});
-customRole.addToPolicy(
-  new iam.PolicyStatement({
-    effect: iam.Effect.ALLOW,
-    resources: ['*'],
-    actions: [
-      'ssm:*',
-    ],
-  }),
-);
-const getParameterNoPolicy = new AwsCustomResource(stack, 'GetParameterNoPolicy', {
-  resourceType: 'Custom::SSMParameter',
-  onUpdate: {
-    service: 'SSM',
-    action: 'getParameter',
-    parameters: {
-      Name: ssmParameter.parameterName,
-      WithDecryption: true,
-    },
-    physicalResourceId: PhysicalResourceId.fromResponse('Parameter.ARN'),
-  },
-  role: customRole,
-});
+    new cdk.CfnOutput(this, 'MessageId', { value: snsPublish.getResponseField('MessageId') });
+    new cdk.CfnOutput(this, 'TopicArn', { value: listTopics.getResponseField('Topics.0.TopicArn') });
+    new cdk.CfnOutput(this, 'ParameterValue', { value: getParameter.getResponseField('Parameter.Value') });
+    new cdk.CfnOutput(this, 'ParameterValueNoPolicy', { value: getParameterNoPolicy.getResponseField('Parameter.Value') });
 
-new cdk.CfnOutput(stack, 'MessageId', { value: snsPublish.getResponseField('MessageId') });
-new cdk.CfnOutput(stack, 'TopicArn', { value: listTopics.getResponseField('Topics.0.TopicArn') });
-new cdk.CfnOutput(stack, 'ParameterValue', { value: getParameter.getResponseField('Parameter.Value') });
-new cdk.CfnOutput(stack, 'ParameterValueNoPolicy', { value: getParameterNoPolicy.getResponseField('Parameter.Value') });
+    if (props?.runtime) {
+      const awsCustomResourceProviderId ='AWS679f53fac002430cb0da5b7982bd2287';
+      const provider = this.node.findChild(awsCustomResourceProviderId).node.defaultChild as lambda.CfnFunction;
+      provider.runtime = props.runtime.name;
+    }
```

Comment:
Lets just import region info and set the default here instead of escape hatching.
```suggestion
    if (props?.runtime) {
      regionInfo.Fact.register({
        region: 'us-east-1', // your region
        name: regionInfo.FactName.DEFAULT_CR_NODE_VERSION,
        value: props.runtime.name,
      }, true);
    }
```

Reply:
fair enough, I do think those are good points.